testbench/common.py

```diff
@@ -662,8 +664,8 @@ def streamer():
                     yield d[r:chunk_end]
                 if bytes_yield >= limit:
                     database.dequeue_next_instruction(test_id, method)
-                    if conn is not None:
-                        conn.close()
+                    if conn_channel is not None:
+                        conn_channel.handle_close()
```

Comment:
Looks like this used to be tested and no longer is.  Are you sure your tests are working as expected?